[2주차] 기본과제

JungYoonShin opened the pull request:
곧 쓰게씀니다..

JungYoonShin committed:
[feat]: 일기 생성 및 상세 조회 로직 구현

JungYoonShin committed:
[feat]: 일기 작성시 글자수 검증 구현

- Validation 어노테이션을 통해, 일기 작성 시 글자 수를 30자 이하로 제한하도록 했습니다.

JungYoonShin committed:
[Feat]: 일기 목록 조회 구현

JungYoonShin committed:
[Feat]: 일기 삭제 및 수정 구현

yxinot commented on src/main/java/org/sopt/seminar2/diary/domain/Diary.java:
```diff
@@ -0,0 +1,50 @@
+import static lombok.AccessLevel.PROTECTED;
+
+@Entity
+@Getter
```
파트장님이 앱잼 전까지는 최대한 @Getter 사용을 지양해보자고 하셔서! 다음부터는 이부분도 신경써주시면 좋을거같아용

JungYoonShin replied:
우옹 해당 부분 신경을 못썼네요.. 감사합니다!!

yxinot commented on src/main/java/org/sopt/seminar2/diary/service/DiaryService.java:
```diff
@@ -0,0 +1,66 @@
+        diaryRepository.delete(findDiary(id));
+    }
+
+    @Transactional
```
여기에 한번 더 @Transactional을 사용하신 이유가 있는지 궁금해요!

JungYoonShin replied:
오..... 잘못 선언했다요...코드가 뭔가 이상하네요 원래 클래스 단에 @Transactional(readOnly = true)를 붙이고 메서드 단에 필요시에 @transactional을 붙이려고 했는데... 덕분에 이상한 걸 알았으요 감삼당!!

yxinot reviewed (approved):
전반적으로 코드가 깔끔하고 간결해서 리뷰하기 수월했습니다 역시 오비.... 👍

hfjxjjd123 commented on src/main/java/org/sopt/seminar2/diary/api/DiaryController.java:
```diff
@@ -0,0 +1,69 @@
+
+    private final DiaryService diaryService;
+
+    @PostMapping("/diary")
+    public ResponseEntity<Void> postDiary(
+            @Valid @RequestBody DiaryCreateRequest diaryCreateRequest
+    ) {
+        diaryService.postDiary(diaryCreateRequest);
+        return ResponseEntity.ok().build();
+    }
```
아! Void로 처리하는 간단한 방법이 있다니....
이런 식으로 처리하면 클라이언트에서는 어떤 형태의 Response를 받게 되나요?
200 OK에 별다른 메시지 없이 전달이 되나요?

JungYoonShin replied:
넵 맞아요 별다른 `response Body`는 없고, `status`만 200으로 받게 됩니다!!

hfjxjjd123 commented on src/main/java/org/sopt/seminar2/diary/api/dto/request/DiaryCreateRequest.java:
```diff
@@ -0,0 +1,11 @@
+
+import jakarta.validation.constraints.Size;
+
+public record DiaryCreateRequest(
+        String title,
```
record 타입이 어떤건지, 썼을 때 어떤 이점이 있는지 궁금해요

ose0221 replied:
저도 처음 보는 거라 한번 찾아보면 좋겠네요

JungYoonShin replied:
먼저 `생성자`, `accessors(getter)` `equals()`, `hashCode()`, `toString()`등을 작성하지 않아도 된다는 장점이 존재합니당
또한 불변 객체여서 `thead-safe`하다는 장점도 존재합니다!

hfjxjjd123 reviewed (approved):
잘 모르는 제가 봐도 깔끔하고 가독성있어서 좋았어요 👍👍

ose0221 commented:
다른 분들 리뷰처럼 정말 가독성이 좋은 것 같습니다! 👍 특히 디렉토리도 깔끔하게 되어 있어서 코드를 찾아보는 데에 훨씬 수월했던 것 같습니다 !

JungYoonShin committed:
[refactor]: service단 @Transactional 적용 범위 재설정

- 클래스단에 @Transactional를 적용하지 않도록 했습니다.

- 단순 조회의 경우(더티체킹x)
  - 지연 로딩이 필요한 경우에만 @Transactional(readOnly = true)를 메서드 단에 적용 ->

- 더티체킹을 사용할 경우
  - @Transactional를 메서드 단에 적용
